packages/clay-color-picker/src/Custom.tsx

```diff
@@ -0,0 +1,240 @@
+		<div className="input-group">
+			<div className="input-group-item">
+				<input
+					className="form-control input-group-inset input-group-inset-before"
+					onChange={event => {
+						const newVal = Number(event.target.value);
+
+						onChange({[name]: newVal});
+					}}
+					type="text"
+					value={value}
+				/>
+				<label className="input-group-inset-item input-group-inset-item-before">
+					{name.toUpperCase()}
+				</label>
+			</div>
+		</div>
+	</div>
+);
+
+interface CustomProps {
+	colors: string[];
+	label?: string;
+	onChange: (val: string) => void;
+	onColorsChange: (val: string[]) => void;
+	spritemap: string;
+	value: string;
+}
+
+/**
+ * Renders the custom color picker
+ */
+const Custom: React.FunctionComponent<CustomProps> = ({
+	colors,
+	label,
+	onChange,
+	onColorsChange,
+	spritemap,
+	value,
+}) => {
+	const color = tinycolor(value);
+
+	const [activeSplotchIndex, setActiveSplotchIndex] = useState(0);
+	const [hue, setHue] = useState(0);
+	const [editorActive, setEditorActive] = useState(false);
+	const [hexInputVal, setHexInput] = useHexInput(color.toHex());
+
+	const {b, g, r} = color.toRgb();
+	const {s, v} = color.toHsv();
+
+	const rgbArr: [number, string][] = [[r, 'r'], [g, 'g'], [b, 'b']];
+
+	const setNewColor = (colorValue: tinycolor.Instance, setInput = true) => {
+		const hexString = colorValue.toHex();
+
+		const newColors = [...colors];
+
+		newColors[activeSplotchIndex] = hexString;
+
+		onColorsChange(newColors);
+
+		onChange(hexString);
+
+		if (setInput) {
+			setHexInput(colorValue.toHex());
+		}
+	};
+
+	return (
+		<React.Fragment>
+			{label && (
+				<div className="clay-color-header">
+					<span className="component-title">{label}</span>
+
+					<button
+						className={`${
+							editorActive ? 'close' : ''
+						} component-action`}
+						onClick={() => setEditorActive(!editorActive)}
+						type="button"
+					>
+						<Icon
+							spritemap={spritemap}
+							symbol={editorActive ? 'times' : 'drop'}
+						/>
+					</button>
+				</div>
+			)}
+
+			<div className="clay-color-swatch">
+				{colors.map((hex, i) => (
+					<div className="clay-color-swatch-item" key={i}>
+						<Splotch
+							active={i === activeSplotchIndex}
+							onClick={() => {
+								if (hex === 'FFFFFF') {
+									setEditorActive(true);
+								}
+
+								setActiveSplotchIndex(i);
+
+								setHue(tinycolor(hex).toHsv().h);
+
+								onChange(hex);
+							}}
+							value={hex}
+						/>
+					</div>
+				))}
+			</div>
+
+			{editorActive && (
+				<React.Fragment>
+					<div className="clay-color-map-group">
+						<GradientSelector
+							color={color}
+							hue={hue}
+							onChange={(saturation, visibility) => {
+								setNewColor(
+									tinycolor({
+										h: hue,
+										s: saturation,
+										v: visibility,
+									})
+								);
+							}}
+						/>
+
+						<div className="clay-color-map-values">
+							{rgbArr.map(([val, key]) => (
```

Comment:
It seems to have some problems when I try to type in the first input:

- Before the value is 0, I type 2 it forces the number to the 255 digit (Because we do not have 02 and this is invalid) but this happens only in the first RGB input.
- When I try to erase the value, it forces the values.

I'm not sure if the best would force or correct the values ​​when the user is typing, maybe doing nothing or adding feedback showing that this RBA is invalid is the best, the experience is bad as long as we want to try to enter a value and it gets forcing.

<img src="http://g.recordit.co/zWzAGuSHvK.gif" />